source/Plugins.cpp

```diff
@@ -50,6 +50,106 @@ namespace {
 
 
 
+// Checks if there are any dependencies of any kind.
+bool Plugin::PluginDependencies::IsEmpty() const
+{
+	return required.empty() && optional.empty() && conflicted.empty();
+}
+
+
+
+// Checks if there are any duplicate dependencies. E.g. the same dependency in both required and conflicted.
+bool Plugin::PluginDependencies::IsValid() const
+{
+	// We will check every dependency before returning to allow the
+	// plugin developer to see all errors and not just the first.
+	bool isValid = true;
+
+	// Required dependencies will already be valid due to sets not
+	// allowing duplicate values. Therefore we only need to check optional
+	// and conflicts.
+	for(const string &dependency : optional)
+	{
+		if(required.count(dependency))
+		{
+			Logger::LogError("Warning: Optional dependency with the name \"" + dependency
+				+ "\" was already found in required dependencies list.");
+		}
+	}
+	for(const string &dependency : conflicted)
+	{
+		if(required.count(dependency))
+		{
+			isValid = false;
+			Logger::LogError("Warning: Conflicts dependency with the name \"" + dependency
+				+ "\" was already found in required dependencies list.");
+		}
+		if(optional.count(dependency))
+		{
+			isValid = false;
+			Logger::LogError("Warning: Conflicts dependency with the name \"" + dependency
+				+ "\" was already found in optional dependencies list.");
+		}
+	}
+
+	return isValid;
+}
+
+
+
+// Constructs a description of the plugin from its name, tags, dependencies, etc.
+string Plugin::CreateDescription() const
+{
+	string text;
+	if(!version.empty())
+	{
+		text += "Version: " + version + '\n';
+	}
+	if(!authors.empty())
+	{
+		text += "Authors: ";
+		for(const string &author : authors)
+			text += author + ',';
```

Comment:
Shouldn't this be comma + space rather than just a comma?
```suggestion
			text += author + ", ";
```